server/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -319,6 +333,7 @@ public <T> T namedObject(Class<T> categoryClass, String name, Object context) th
                     }
                 }
                 T namedObject = getXContentRegistry().parseNamedObject(categoryClass, name, this, context);
+                queryNameConsumer.accept(name);
```

Comment:
Does this need to be inside the `if` statement? Otherwise we'll track named objects that aren't queries here, right?

Reply:
oh boy, yes you are right. I will think of a way to test that, I guess there aren't a lot of other named objects around.

Reply:
decay functions are, and I added a test that failed before moving the accept call and succeeds after. Thanks for finding this.